rmgpy/data/base.py

```diff
@@ -875,10 +875,10 @@ def matchNodeToChild(self, parentNode, childNode):
             return False
         
         #If the parentNode is a Group and the childNode is a LogicOr there is nothing to check,
-        #so it gets an automatic pass. However, we do need to check that everything down this
+        #except that the parent is listed in the attributes. However, we do need to check that everything down this
         #family line is consistent, which is done in the databaseTest unitTest
         elif isinstance(parentNode.item, Group) and isinstance(childNode.item, LogicOr):
-            return True
+            return childNode.parent is parentNode
```

Comment:
in order to spot possible suspicious parent/child, for this if situation `isinstance(parentNode.item, Group) and isinstance(childNode.item, LogicOr)`, we  might also want to check the items in the `LogicOr` are children of the `parentNode`